modules/core/src/main/java/org/apache/ignite/internal/manager/IgniteComponent.java

```diff
@@ -28,12 +30,13 @@ public interface IgniteComponent {
      * thread pools and threads goes here.
      *
      * <p>All actions in the component startup is divided into two categories: sync actions,
-     * that can be executed synchronously in order for the component to be usable by other components during their startup,
-     * and async actions, that are wrapped in a CompletableFuture and returned from the start method.
+     * that can be executed synchronously in order for the component to be usable by other components during their startup, and async
+     * actions, that are wrapped in a CompletableFuture and returned from the start method.
      *
+     * @param startupExecutor The executor that will execute the async part of start.
      * @return Future that will be completed when the asynchronous part of the start is processed.
      */
-    CompletableFuture<Void> startAsync();
+    CompletableFuture<Void> startAsync(ExecutorService startupExecutor);
```

Comment:
I suggest to pass something like ComponentContext instead of executor, whilch provides executorservice and whatever we want later.